langchain/src/chat_models/base.ts

```diff
@@ -80,17 +81,17 @@ export abstract class BaseChatModel extends BaseLanguageModel {
       options: parsedOptions,
       invocation_params: this?.invocationParams(),
     };
-    const runManager = await callbackManager_?.handleChatModelStart(
+    const runManagers = await callbackManager_?.handleChatModelStart(
       this.toJSON(),
       messages,
       undefined,
       undefined,
       extra
     );
     try {
-      const results = await Promise.all(
+      results = await Promise.all(
         messages.map((messageList) =>
-          this._generate(messageList, parsedOptions, runManager)
+          this._generate(messageList, parsedOptions, runManagers?.[0])
```

Comment:
this should be `i` instead of 0

Reply:
It actually doesn't matter which one we pass as the handlers are all the same. I guess using i is more idiomatic though